### ISSUE TITLE
False positive regex warnings on constant

### ISSUE THREAD

**Issue by caricare2** [kalessil/phpinspectionsea#2018]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | Php Inspections (EA Extended) 6.3.0 |
| Language level |PHP 8.4    |

Current behaviour
--
I noticed two unexpected errors displayed in PHPStorm with the following constant:
`private const string BASE_URL = '/a/categories';`

[EA] 'e' modifier is deprecated, please use 'preg_replace_callback()' instead.
[EA] /u modifier is missing (r modifier found).

This went away when I changed the URL to `/a/cat` instead.

It seems to be triggered by this line (although the error is on the constant line):
```
        self::assertMatchesRegularExpression(
            '#^' . \preg_quote(self::BASE_URL, '#') . '/\d+$#',
            $location,
            'Expected to be redirected to the saved category'
        );
```

But preg_quote is not using it as a regex, just quoting the regex characters, so I don't understand why this would trigger the warning.

Expected behaviour
--
No error should be displayed

Environment details
---
PhpStorm 2026.1.4
MacOS Tahoe 26.2
